pyumi/umi_project.py

```diff
@@ -541,6 +554,17 @@ def try_make_geom(series, height_column_name):
                 f"{gdf.size} breps created in "
                 f"{time.time() - start_time:,.2f} seconds"
             )
+        gdf_error.explode()
+        gdf_error["Brep Error"] = [list() for x in range(len(gdf_error.index))]
+        # Iterate over the rows of the error gdf since its shape is different
+        # This allows you to report the original building index and
+        # which part has an error while following the original shape
+        for index, row in gdf_error.iterrows():
```

Comment:
I see you use a lot of for loops with iterrows. You can simply use DataFrame.apply(lambda row: **do something with the row data**, axis=1). Axis=1 specifies a row by row method. The returned value will be a Series that has the same index as the original dataframe such that df["A new Column"] = df.apply(lambda row: **do something with the row data**, axis=1) is super simple to implement